bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/DeviceAttributes.java

```diff
@@ -45,4 +48,16 @@
     public @NonNull Object getFieldsOf() {
         return this;
     }
+
+    /**
+     * Return a map with all field values.
+     */
+    public Map<String, Object> asMap() {
```

Comment:
Hm. I don't like this method. You are using the keys only and generally I'm using Java reflection (we are in java we should use that fact) via GSon etc. Could you use reflection here as well, please? (Getting all fields that are not final or transient).

Reply:
Hi! I have created the same method that was available in the PropertyAttributes class for the DeviceAttributes and NodeAttributes classes, for the sake of symmetry. Do you want me to remove this asMap() method and use the "getClass().getDeclaredFields()" method in the getRetainedTopics() function?

Reply:
Yes that `asMap` in other classes was not introduced by me and I'm going to remove it at some point. `getDeclaredFields` yup